packages/flutter/lib/src/cupertino/sheet.dart

```diff
@@ -153,52 +161,69 @@ Future<T?> showCupertinoSheet<T>({
   )
   WidgetBuilder? pageBuilder,
   WidgetBuilder? builder,
+  ScrollableWidgetBuilder? scrollableBuilder,
   bool useNestedNavigation = false,
   bool enableDrag = true,
 }) {
-  assert(pageBuilder != null || builder != null);
+  assert(pageBuilder != null || builder != null || scrollableBuilder != null);
+  assert((pageBuilder == null && builder == null) || scrollableBuilder == null);
 
-  final WidgetBuilder? effectivePageBuilder = builder ?? pageBuilder;
-  final WidgetBuilder widgetBuilder;
+  final WidgetBuilder? effectiveBuilder = builder ?? pageBuilder;
   final nestedNavigatorKey = GlobalKey<NavigatorState>();
   if (!useNestedNavigation) {
-    widgetBuilder = effectivePageBuilder!;
-  } else {
-    widgetBuilder = (BuildContext context) {
-      return NavigatorPopHandler(
-        onPopWithResult: (T? result) {
-          nestedNavigatorKey.currentState!.maybePop();
+    final PageRoute<T> route = effectiveBuilder != null
+        ? CupertinoSheetRoute<T>(builder: effectiveBuilder, enableDrag: enableDrag)
+        : CupertinoSheetRoute<T>.scrollable(
+            scrollableBuilder: scrollableBuilder,
+            enableDrag: enableDrag,
+          );
+
+    return Navigator.of(context, rootNavigator: true).push<T>(route);
+  }
+
+  Widget nestedNavigationContent(WidgetBuilder builder) {
+    return NavigatorPopHandler(
+      onPopWithResult: (T? result) {
+        nestedNavigatorKey.currentState!.maybePop();
+      },
+      child: Navigator(
+        key: nestedNavigatorKey,
+        initialRoute: '/',
+        onGenerateInitialRoutes: (NavigatorState navigator, String initialRouteName) {
+          return <Route<void>>[
+            CupertinoPageRoute<void>(
+              builder: (BuildContext context) {
+                return PopScope(
+                  canPop: false,
+                  onPopInvokedWithResult: (bool didPop, Object? result) {
+                    if (didPop) {
+                      return;
+                    }
+                    Navigator.of(context, rootNavigator: true).pop(result);
```

Comment:
I know this is not introduced by this pr, but what if there is multiple navigator above the nested navigator? I thought we may want to pop just one level above this nested navigator

Reply:
`showCupertinoSheet` always pushes to the root navigator, especially as it needs to slide the whole page down. So popping the sheet also targets the root navigator, as the last route in the stack should theoretically be the sheet you are meaning to pop, in the case of nested navigation.

If there is a situation where that does not work for their app, then we direct people to use `CupertinoSheetRoute` directly instead of `showCupertinoSheet`.

Reply:
Ah ok, makes sense